apps/dashboard/src/components/workflow-editor/steps/email/maily.tsx

```diff
@@ -74,11 +113,37 @@ export const Maily = (props: MailyProps) => {
                   }}
                 />
               </FormControl>
+              <FormMessage />
             </div>
-            <FormMessage />
           </>
         );
       }}
     />
   );
 };
+
+const dedupAndSortVariables = (
+  variables: { name: string; required: boolean }[],
+  query: string
+): { name: string; required: boolean }[] => {
+  // Filter variables that match the query
+  const filteredVariables = variables.filter((variable) => variable.name.toLowerCase().includes(query.toLowerCase()));
+
+  // Deduplicate based on name property
+  const uniqueVariables = Array.from(new Map(filteredVariables.map((item) => [item.name, item])).values());
+
+  // Sort variables: exact matches first, then starts with query, then alphabetically
+  return uniqueVariables.sort((a, b) => {
```

Comment:
Is this algorithm the same one that Maily used internally? I am curious if we need a more sophisticated string comparison/fuzzy search library to match query with results like https://www.npmjs.com/package/bloodhound-js.

Reply:
looks good to me as is, but we can improve.